pkg/controllers/management/auth/globalroles/enqueue_test.go

```diff
@@ -0,0 +1,464 @@
+package globalroles
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/wrangler/pkg/generic/fake"
+	"github.com/rancher/wrangler/pkg/relatedresource"
+	"github.com/stretchr/testify/require"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func Test_enqueueGRBs(t *testing.T) {
```

Comment:
Feel free to start each of the three tests in parallel.
A test can just start with `t.Parallel`, unless there is some shared state among tests, which is rare and should be avoided anyways.

Reply:
The sub-tests are marked as parallel - https://github.com/rancher/rancher/pull/42319/files#diff-532b8642b474effd781a771f2ed4a9cd0ca119e02219c12e62454e936ecfae47R119 which I think is the correct place for this. Can you elaborate on why you would want the overaching test marked this way?

Reply:
Right, you can have the test itself execute in parallel in relation to other tests. Subtests within each test can also be run in parallel, so having both is best.
https://engineering.mercari.com/en/blog/entry/20220408-how_to_use_t_parallel/

Reply:
That makes sense, thanks for explaining. I've made the changes to accommodate.